cylc/flow/parsec/jinja2support.py

```diff
@@ -182,9 +186,78 @@ def jinja2environment(dir_=None):
     env.globals['environ'] = os.environ
     env.globals['raise'] = raise_helper
     env.globals['assert'] = assert_helper
+
+    # Insert key/value pairs from rose-suite.conf into jinja2 env.
+    from ast import literal_eval
+    rose_vars = get_rose_vars(dir_)
+    if rose_vars is not None:
+        for key, value in rose_vars.items():
+            env.globals[key] = literal_eval(value)
```

Comment:
Shouldn't this belong in `template_vars` rather then environment globals?

Reply:
This would probably make more sense - I'll give it a go.

Reply:
Why? These are suite variables, so logically serve the same purpose as Cylc template vars? Plus, that way you won't need to solve it separately for EmPy.

Reply:
I think the easiest place to implement this centrally would be in [`cylc.flow.parsec.fileparse.read_and_proc`](https://github.com/cylc/cylc-flow/blob/218fa6d/cylc/flow/parsec/fileparse.py#L203-L269). All the relevant info is there, you can even test if you need to load from `jinja2:suite.rc` or `empy:suite.rc`.

BTW: Any plans to rename `jinja2:suite.rc` & `empy:suite.rc` into something more in line with the current Cylc naming? Given `suite.rc` -> `cylc.flow` rename it would only make sense.

Reply:
We need to back support both `jinja2:suite.rc`, however, the two can be merged and called something like "template_variables", but not expecting all of these changes in the first PR.

Reply:
Turns out my problem was that we don't _really_ want to mix "CLI" template variables, which must be serializable because they are stored in the suite DB with "File" template variables, which are stored in a file and do not need to be stored in the suite DB: Unless we force a massive change on users the file template variables must be potentially unserializable, so we mustn't add them to the dictionary to be stored in the DB. Fortunately, since we store them in a file this isn't a problem.